models/restaurants.go

```diff
@@ -1,27 +1,41 @@
 package models
 
+import "fmt"
+
+// Restaurant data struct
 type Restaurant struct {
 	ID   string `json:"id"`
 	Name string `json:"name"`
 }
 
-func AllRestaurants() ([]*Restaurant, error) {
-	rows, err := db.Query("SELECT * FROM restaurants")
+// AllRestaurants return all the restaurant from database
+func AllRestaurants() ([]Restaurant, error) {
+	stmt, err := db.Prepare("SELECT id, name FROM restaurants")
+	if err != nil {
+		fmt.Printf(err.Error())
+		return nil, err
+	}
+	defer stmt.Close()
+
+	rows, err := stmt.Query()
 	if err != nil {
+		fmt.Printf(err.Error())
 		return nil, err
 	}
-	defer rows.Close()
 
-	rts := make([]*Restaurant, 0)
+	rts := make([]Restaurant, 0)
 	for rows.Next() {
-		rt := new(Restaurant)
+		rt := Restaurant{}
 		err := rows.Scan(&rt.ID, &rt.Name)
 		if err != nil {
+			fmt.Printf(err.Error())
 			return nil, err
 		}
 		rts = append(rts, rt)
 	}
+
 	if err = rows.Err(); err != nil {
+		fmt.Printf(err.Error())
```

Comment:
what does rows.Err() do?

Reply:
it reports last error encountered by `rows.Scan()`

https://golang.org/pkg/database/sql/#example_DB_QueryContext